### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by shkhaliq**
Adds support for Swift Package Manager
Added steps for installation in the README
Added badge for the version of the cocoapod

### Concerns

* I made changes to the `codableFromFile` method and the reasoning was so that I could run tests from the commandline `swift test` but the are sources that are copied over using bundle resources but Swift PM doesn't support it.
  * I am happy to remove those changes if we don't want to support SPM going forward

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+    ```swift
+    import PackageDescription
+
+    let package = Package(
+      name: "MyAwesomeApp",
+      dependencies: [
+        .Package(url: "https://github.com/nerdishbynature/octokit.swift", from: "0.8.0"),
+      ]
+    )
+    ```
```
Comment (phatblat) [inline README.md]: Using both indentation and backticks is redundant. I suggest removing the indentation in favor of the swift code block, plus it will be a little easier to keep this updated.

**Comment by hariskhaliq** [comment]
@phatblat Done

File: OctoKitTests/TestHelper.swift [hunk 2]
```diff
@@ -22,8 +22,15 @@ internal class Helper {
     }
 
     internal class func codableFromFile<T>(_ name: String, type: T.Type) -> T where T: Codable {
+        var path: String
         let bundle = Bundle(for: self)
-        let path = bundle.path(forResource: name, ofType: "json")!
+        if let bundlePath = bundle.path(forResource: name, ofType: "json") {
```
Comment (pietbrauer) [inline OctoKitTests/TestHelper.swift]: Why was this changed? The remaining test are untouched.

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+let package = Package(
+    name: "OctoKit",
+
+    products: [
+        .library(
+            name: "OctoKit",
+            targets: ["OctoKit"]),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/nerdishbynature/RequestKit.git", .branch("master")),
```
[the remark this answers is not in the document]
Comment (pietbrauer) [reply Package.swift]: Version 2.3.0 is released now.

File: OctoKitTests/TestHelper.swift (hunk 2 above)
Comment (shkhaliq) [reply OctoKitTests/TestHelper.swift]: @pietbrauer I added some reasoning in the PR description. Here is the snippet:

> * I made changes to the `codableFromFile` method and the reasoning was so that I could run tests from the commandline `swift test` but there are sources that are copied over using bundle resources but Swift PM doesn't support it.
> * I am happy to remove those changes if we don't want to support SPM going forward

Comment (pietbrauer) [reply OctoKitTests/TestHelper.swift]: Ah, I see. Sorry for not remembering it.
